frontend/src/tests/posting-tests.js

```diff
@@ -1,39 +1,354 @@
+            intro_text: "Posting for CSC400F",
+            open_date: "2021/04/01",
+            close_date: "2021/05/01",
+            availability: "auto",
+        };
+
+        resp = await apiPOST(
+            `/admin/sessions/${session.id}/postings`,
+            newPosting
+        );
+        expect(resp).toHaveStatus("success");
+
+        const newPosting2 = {
+            name: "CSC400F TA",
+            intro_text: "Posting for CSC400F",
+            open_date: "2021/04/01",
+            close_date: "2021/05/01",
+            availability: "auto",
+        };
+        resp = await apiPOST(
+            `/admin/sessions/${newSession.id}/postings`,
+            newPosting2
+        );
+        expect(resp).toHaveStatus("success");
+    });
+
+    it("Create a posting_position for a posting", async () => {
+        resp = await apiPOST(
+            `/admin/postings/${posting.id}/posting_positions`,
+            { ...postingPos, position_id: position.id }
+        );
+        expect(resp).toHaveStatus("success");
+        checkPropTypes(postingPositionPropTypes, resp.payload);
+        expect(resp.payload.posting_id).toEqual(posting.id);
+        Object.assign(postingPos, resp.payload);
     });
 
-    it.todo("Create a posting for a session");
-    it.todo("Modify a posting");
-    it.todo("Cannot set the `status` of a posting to an invalid value");
-    it.todo("Fetch all postings for a session");
-    it.todo(
-        "Can set `custom_questions` to an arbitrary serializable object and the same object (i.e., an object and not the stringified version) gets returned."
-    );
-    it.todo("Two postings for the same session cannot have the same name");
-    it.todo("Two postings for different sessions may have the same name");
-    it.todo("Create a posting_position for a posting");
-    it.todo("A posting contains a list of all associated posting_position ids");
-    it.todo("A posting contains a list of all associated application ids");
-    it.todo("Fetch a survey for a posting");
-    it.todo(
-        "Survey for a posting includes questions related to each PostingPosition"
-    );
-    it.todo(
-        "Cannot create a posting_position with a position associated with a different session than the posting"
-    );
-    it.todo("Delete a posting_position");
-    it.todo("Modify a posting_position");
-    it.todo("Deleting a posting also deletes all associated posting_positions");
-    it.todo(
-        "Cannot create two posting_positions with the same position_id for a single posting"
-    );
-    it.todo(
-        "Can create two posting_positions with the same position_id for different postings"
-    );
+    // `/admin/postings/${posting.id}/applications` need to be implemented
+    it.todo("Fetch all applications associated with a posting");
+
+    it("Fetch a survey for a posting", async () => {
+        resp = await apiGET(`/admin/postings/${posting.id}/survey`);
+        expect(resp).toHaveStatus("success");
```

Comment:
you may want to also check `checkPropTypes` on the `survey` object, and probably by first defining a correct `survey ` prop type in `frontend\src\api\defs\prop-types.ts`